gcp/gce_instance.go

```diff
@@ -0,0 +1,85 @@
+package gcp
+
+import (
+	"errors"
+	"strings"
+	"time"
+)
+
+func zoneFromUrl(url string) (string, error) {
```

Comment:
Add unit tests for this method. It's important that we know it works correctly so we don't nuke the wrong zone/region.